backend/models/member.go

```diff
@@ -0,0 +1,89 @@
+package models
+
+import (
+	"errors"
+
+	"github.com/go-playground/validator/v10"
+	"gorm.io/gorm"
+)
+
+// Member struct has Member table's columns
+type Member struct {
+	ID        int    `json:"id"`
+	UserID    string `json:"user_id"  gorm:"not null;" validate:"required"`
+	ProjectID int    `json:"project_id"  gorm:"not null" validate:"required"`
+	Admin     bool   `json:"admin"`
+}
+
+var ErrAccessDenied = errors.New("error access denied")
+
+// Validate validates the comment struct using the validate tag
+func (member *Member) Validate() error {
+	validate := validator.New()
+	return validate.Struct(member)
+}
+
+// CreateNewMember adds a new member to member table
+func (db *DBClient) CreateNewMember(member Member, userId string) error {
+	var authMember Member
+	rows := db.Client.Model(&Member{}).Where("user_id = ? AND project_id = ?", userId, member.ProjectID).First(&authMember)
+	if !authMember.Admin {
+		var project Project
```

Comment:
no need to get the project. if the creator is not an admin return `ErrAccessDenied` directly
same for update